src/health.py

```diff
@@ -0,0 +1,152 @@
+#!/usr/bin/env python3
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Manager for handling Kafka machine health."""
+
+import json
+import logging
+import subprocess
+from statistics import mean
+from typing import Tuple
+
+from ops.framework import Object
+
+from literals import SNAP_NAME, Status
+
+logger = logging.getLogger(__name__)
+
+
+class KafkaHealth(Object):
+    """Manager for handling Kafka machine health."""
+
+    def __init__(self, charm) -> None:
+        super().__init__(charm, "kafka_health")
+        self.charm = charm
+
+    @property
+    def _service_pid(self) -> int:
+        """Gets most recent Kafka service pid from the snap logs."""
+        return self.charm.snap.get_service_pid(service_name=SNAP_NAME)
+
+    def _get_current_memory_maps(self) -> int:
+        """Gets the current number of memory maps for the Kafka process."""
+        return int(
+            subprocess.check_output(
+                f"sudo cat /proc/{self._service_pid}/maps | wc -l",
```

Comment:
I am not sure if sudo is needed. can we test without it?

Reply:
I believe it does, let me know what you think:

```bash
root@juju-0b8d7f-1:~# cat /proc/11967/maps | wc -l
562

ubuntu@juju-0b8d7f-1:~$ cat /proc/11967/maps | wc -l
cat: /proc/11967/maps: Permission denied
0
```

Reply:
It does not when ran through the charm, you're right, because charms run as sudo anyway.
I've skipped these checks in the unit-tests however as we'd need sudo to access the files I believe.